### PR TITLE
Fixed issue #16903: Windows server : unable to save attachements

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue #16905: Windows server : good attachment throw JS error (and broke all JS action)
Dev: save only basename : JS don't care of separator
Dev: save only basename, find files in survey upload dir (always)

### CODE REVIEW

**Comment by olleharstedt** [comment]
Why is this fix so big? :cry:

File: application/controllers/admin/emailtemplates.php [hunk 1]
```diff
@@ -99,38 +99,22 @@ function index($iSurveyId)
      */
     function update($iSurveyId)
     {
-        $sBaseUrl = Yii::app()->getBaseUrl();
-        $uploadUrl = Yii::app()->getConfig('uploadurl');
-        if (substr($uploadUrl, 0, strlen($sBaseUrl)) == $sBaseUrl) {
-            $uploadUrl = substr($uploadUrl, strlen($sBaseUrl));
-        }
-        $sBaseAbsoluteUrl = Yii::app()->getBaseUrl(true);
-        $sPublicUrl = Yii::app()->getConfig("publicurl");
```
Comment (olleharstedt) [inline application/controllers/admin/emailtemplates.php:108]: Remove use of publicurl?

**Comment by Shnoulle** [comment]
> Why is this fix so big? cry

Because there are no helper or model system to get and fix attachements : 

The «only fix» was using `$baseName = pathinfo($attachment['url'], PATHINFO_BASENAME);` but it must be done each time we try to find the file , then each time we send a message … with attachment.

**Comment by Shnoulle** [comment]
2 mores ! https://github.com/LimeSurvey/LimeSurvey/pull/1800/commits/9dcf41e2b7c649f65d90c01b7150e36de5b06cb9

I lost admin notification on 1st commit

**Comment by Shnoulle** [comment]
Create a getter SurveyLanguageSetting->getAttachement return the arry is surely the better solution , but it's rewrite, not fix …

**Comment by Shnoulle** [comment]
Oh,

I got aniother idea : manage : use only filename, but save with complete directory . Then less update.

And create a clean getter in 4X

**Pull request closed without merging**